packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -20,6 +20,54 @@ export namespace externalAppAuthentication {
    */
   export type IOriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
 
+  /**
+   * @hidden
+   * Parameters for the authentication pop-up
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type AuthenticatePopUpParameters = {
+    /**
+     * The URL for the authentication pop-up.
+     */
+    url: URL;
+    /**
+     * The preferred width for the pop-up. This value can be ignored if outside the acceptable bounds.
+     */
+    width: number;
+    /**
+     * The preferred height for the pop-up. This value can be ignored if outside the acceptable bounds.
+     */
+    height: number;
+    /**
+     * Some identity providers restrict their authentication pages from being displayed in embedded browsers (e.g., a web view inside of a native application)
+     * If the identity provider you are using prevents embedded browser usage, this flag should be set to `true` to enable the authentication page specified in
+     * the {@link url} property to be opened in an external browser.
+     * If this flag is `false`, the page will be opened directly within the current hosting application.
+     *
+     * This flag is ignored when the host for the application is a web app (as opposed to a native application) as the behavior is unnecessary in a web-only
+     * environment without an embedded browser.
+     */
+    isExternal: boolean;
+  };
+
+  /**
+   * @hidden
+   * Parameters for SSO authentication
```

Comment:
Nit: we should probably clarify that these interfaces are used exclusively with the externalApp authentication APIs.